src/hugr/views/sibling.rs

```diff
@@ -215,6 +208,185 @@ where
+
+    fn node_count(&self) -> usize {
+        // Same as SiblingGraph.
+        self.base_hugr().hierarchy.child_count(self.root.index) + 1
+    }
+
+    fn edge_count(&self) -> usize {
+        // Same as SiblingGraph
+        self.nodes()
+            .map(|n| self.output_neighbours(n).count())
+            .sum()
+    }
+
+    fn nodes(&self) -> Self::Nodes<'_> {
+        // Same as SiblingGraph
+        let children = self
+            .base_hugr()
+            .hierarchy
+            .children(self.root.index)
+            .map_into();
+        iter::once(self.root).chain(children)
+    }
+
+    fn node_ports(&self, node: Node, dir: Direction) -> Self::NodePorts<'_> {
+        match self.contains_node(node) {
+            true => self.base_hugr().node_ports(node, dir),
+            false => <FlatRegionGraph as PortView>::NodePortOffsets::default().map_into(),
+        }
+    }
+
+    fn all_node_ports(&self, node: Node) -> Self::NodePorts<'_> {
+        match self.contains_node(node) {
+            true => self.base_hugr().all_node_ports(node),
+            false => <FlatRegionGraph as PortView>::NodePortOffsets::default().map_into(),
+        }
+    }
+
+    fn linked_ports(&self, node: Node, port: Port) -> Self::PortLinks<'_> {
+        match self.contains_node(node) {
+            true => self.base_hugr().linked_ports(node, port),
+            false => EMPTY_GRAPH
+                .port_links(portgraph::PortIndex::new(1))
+                .with_context(self.base_hugr())
+                .map_with_context(|_port, _ctx| panic!("Never called")),
+        }
+    }
+
+    fn node_connections(&self, node: Node, other: Node) -> Self::NodeConnections<'_> {
+        match self.contains_node(node) && self.contains_node(other) {
+            true => self.base_hugr().node_connections(node, other),
+            false => EMPTY_GRAPH
+                .get_connections(portgraph::NodeIndex::new(1), portgraph::NodeIndex::new(2))
+                .with_context(self.base_hugr())
+                .map_with_context(|_port, _ctx| panic!("Never called")),
+        }
+    }
+
+    fn num_ports(&self, node: Node, dir: Direction) -> usize {
+        match self.contains_node(node) {
+            true => self.base_hugr().num_ports(node, dir),
+            false => 0,
+        }
+    }
+
+    fn children(&self, node: Node) -> Self::Children<'_> {
+        match node == self.root {
+            true => self.base_hugr().hierarchy.children(node.index).map_into(),
+            false => portgraph::hierarchy::Children::default().map_into(),
+        }
+    }
+
+    fn neighbours(&self, node: Node, dir: Direction) -> Self::Neighbours<'_> {
+        self.valid_node(node).unwrap(); // ?? Or return empty iterator?
+        self.base_hugr().neighbours(node, dir) // Or self.hugr ?
+    }
+
+    fn all_neighbours(&self, node: Node) -> Self::Neighbours<'_> {
+        self.valid_node(node).unwrap(); // ?? Or return empty iterator?
+        self.base_hugr().all_neighbours(node)
```

Comment:
I'd say return an empty one.

Reply:
Ooops, missed those when I fixed a bunch of other `unwrap`s. Done, thanks! :)